src/js/page.js

```diff
@@ -162,8 +162,8 @@ export default {
         this.$toc
             .on( 'click', 'a', function() {
                 if ( !that.form.pageNavigationBlocked ) {
-                    const index = $( this.parentNode ).prevAll().length;
-                    that.flipToPageContaining( $( that.tocItems[ index ].pageEl ) );
+                    const currentIndex = that.$toc.find( '[role="pageLink"]' ).index( $( this.parentNode ) );
+                    that.flipToPageContaining( $( that.tocItems[ currentIndex ].pageEl ) );
```

Comment:
Sorry, I didn't notice this before. This only flips to the page containing the ToC element. We can perhaps do even better than that by using that.form.goToTarget(questionOrGroup). You may have to build a map of tocItem -> element.